### PR TITLE
[feature] Update xar installer

### PR DESCRIPTION
**Pull request by agh2342**
Handle transition from xars without commit-id or condig-id.

### CODE REVIEW

File: templates/install-xars.xql.j2 [hunk 1]
```diff
@@ -126,7 +126,10 @@ declare function local:deploy-xar($xar) {
             let $log := util:log('info', "install-xars: forcing uninstall of existing version " || $oldversion || ", uninstalling " || $name)
             let $undep := repo:undeploy($name)
             return repo:remove($name)
-        ) else if ((not(empty($newcommitid)) or not(empty($newconfigid))) and (not($newcommitid = $oldcommitid) or not($newconfigid = $oldconfigid))) then (
+        ) else if (
+            (not(empty($newcommitid)) and (not($newcommitid = $oldcommitid))) or
+            (not(empty($newconfigid)) and (not($newconfigid = $oldconfigid)))
```
Comment (joewiz) [inline templates/install-xars.xql.j2]: Consider `exists()` as a more direct form of `not(empty())`, and `eq` (for atomic value comparisons) is preferable to `=` (for sequence comparisons)

**Review by chakl — APPROVED** [review]
ok. please consider @joewiz 's suggestions

File: templates/install-xars.xql.j2 (hunk 1 above)
Comment (joewiz) [reply templates/install-xars.xql.j2]: I should have added, in the same spirit, that a more direct form of even`not($newcommitid eq $oldcommitid)` would be `$newcommitid ne $oldcommitid`.

Comment (agh2342) [reply templates/install-xars.xql.j2]: > Consider `exists()` as a more direct form of `not(empty())`, and `eq` (for atomic value comparisons) is preferable to `=` (for sequence comparisons)

thanks for the feedback. updated PR accordingly.

Comment (agh2342) [reply templates/install-xars.xql.j2]: > I should have added, in the same spirit, that a more direct form of even`not($newcommitid eq $oldcommitid)` would be `$newcommitid ne $oldcommitid`.

I have to disagree here. `($newcommitid ne $oldcommitid)` is NOT equivalent to `not($newcommitid eq $oldcommitid)`.
As per https://www.w3.org/TR/xquery-31/#id-value-comparisons 3.7.1.2 one empty atomized operand makes the comparison return an empty sequence (independent of the operator used).
This empty sequence will evaluate to false() and results in branch not taken.
The branch will only be taken if `$newcommitid` is non-empty. But the branch must be taken if `$oldcommitid` is different from `$newcommitid` OR empty. Using `not(... eq ...)` shows that behaviour.
Do you agree?

Comment (joewiz) [reply templates/install-xars.xql.j2]: You're correct that the two expressions (`not($a eq $b)` and `$a ne $b`) can produce different values, but if either operand is false, the empty sequence produced by `$a ne $b` has an *effective boolean value* of `false()` - and thus the conditional will evaluate to `false()`. For example: `if (1 ne ()) then "yes" else "no"` returns `"no"`. Thus, the result of the conditional is the same regardless of whether you formulate it as `not+eq` or `ne`.

**Pull request merged**